plugins/file-open-save-new/core/src/test/java/org/pentaho/di/plugins/fileopensave/extension/FileOpenSaveExtensionPointTest.java

```diff
@@ -0,0 +1,182 @@
+    ProviderService mockProviderService = mock( ProviderService.class );
+    VFSFileProvider mockVFSFileProvider = mock( VFSFileProvider.class );
+    when( mockProviderService.get( VFSFileProvider.TYPE ) ).thenReturn( mockVFSFileProvider );
+    when( mockVFSFileProvider.isSupported(  vfsPath ) ).thenReturn( true );
+
+    FileOpenSaveExtensionPoint testInstance = new FileOpenSaveExtensionPoint( mockProviderService, null );
+
+    FileDialogOperation opVfs = new FileDialogOperation( FILE_OP_DUMMY_COMMAND  );
+    opVfs.setProvider( null );
+    opVfs.setPath( vfsPath );
+
+    // EXECUTE
+    testInstance.resolveProvider( opVfs );
+
+    // VERIFY
+    assertEquals( VFSFileProvider.TYPE, opVfs.getProvider() );
+  }
+
+  @Test
+  public void testResolveProvider_Repository() throws Exception  {
+    // SETUP
+    ProviderService mockProviderService = mock( ProviderService.class );
+    VFSFileProvider mockVFSFileProvider = mock( VFSFileProvider.class );
+    when( mockProviderService.get( VFSFileProvider.TYPE ) ).thenReturn( mockVFSFileProvider );
+    when( mockVFSFileProvider.isSupported( any() ) ).thenReturn( false );
+    Spoon mockSpoon = mock( Spoon.class );
+    when( mockSpoon.getRepository() ).thenReturn( null );
+    Repository mockRepository = mock( Repository.class );
+    when( mockSpoon.getRepository() ).thenReturn( mockRepository );
+    FileOpenSaveExtensionPoint testInstance = new FileOpenSaveExtensionPoint( mockProviderService, () -> mockSpoon );
+
+    FileDialogOperation opRepository = new FileDialogOperation( FILE_OP_DUMMY_COMMAND  );
+    opRepository.setProvider( null );
+    opRepository.setPath( "//home/randomUser/randomFile.rpt" );
+
+    // EXECUTE
+    testInstance.resolveProvider( opRepository );
+
+    // VERIFY
+    assertEquals( RepositoryFileProvider.TYPE, opRepository.getProvider() );
+  }
+
+  @Test
+  public void testResolveProvider_Local() throws Exception  {
+    // SETUP
+    ProviderService mockProviderService = mock( ProviderService.class );
+    VFSFileProvider mockVFSFileProvider = mock( VFSFileProvider.class );
+    when( mockProviderService.get( VFSFileProvider.TYPE ) ).thenReturn( mockVFSFileProvider );
+    when( mockVFSFileProvider.isSupported( any() ) ).thenReturn( false );
+    Spoon mockSpoon = mock( Spoon.class );
+    when( mockSpoon.getRepository() ).thenReturn( null );
+    FileOpenSaveExtensionPoint testInstance = new FileOpenSaveExtensionPoint( mockProviderService, () -> mockSpoon );
+
+    FileDialogOperation opLocal = new FileDialogOperation( FILE_OP_DUMMY_COMMAND  );
+    opLocal.setProvider( null );
+    opLocal.setPath( "/someUser/someUnixFile" );
+
+    // EXECUTE
+    testInstance.resolveProvider( opLocal );
+
+    // VERIFY
+    assertEquals( LocalFileProvider.TYPE, opLocal.getProvider() );
+  }
+
+  @Test
+  public void testIsVfsPath_nullFileProvider() throws Exception {
+    // SETUP
+    String vfsPath = "pvfs://someConenction/someFilePath";
```

Comment:
No need to fix.
Typo in `someConenction`, here and elsewhere.

Reply:
addressed in https://github.com/pentaho/pentaho-kettle/commit/8d2d79b16445a6dfdaf944f34a702997b3bfe6b5